desc/objectives/_geometry.py

```diff
@@ -459,6 +459,290 @@ def compute(self, params, constants=None):
+        self._data_keys = data_keys
+        super().__init__(
+            things=[coil],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        # local import to avoid circular import
+        from desc.coils import CoilSet, MixedCoilSet
+
+        is_mixed_coils = False
+        if isinstance(self.things[0], CoilSet):
+            if isinstance(self.things[0], MixedCoilSet):
+                is_mixed_coils = True
+                coil = self.things[0].coils
+                self._dim_f = len(coil)
+            else:
+                coil = self.things[0].coils[0]
+                self._dim_f = 1
+        else:
+            coil = self.things[0]
+            self._dim_f = 1
+
+        if self._grid is None:
+            # TODO: raise error if grid, transforms are not the same size as mixed coils
+            self._grid = (
+                LinearGrid(N=32)
+                if not is_mixed_coils
+                else [LinearGrid(N=32)] * self._dim_f
+            )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        if is_mixed_coils:
+            transforms = [
+                get_transforms(self._data_keys, obj=coil[i], grid=self._grid[i])
+                for i in range(self._dim_f)
+            ]
+        else:
+            transforms = get_transforms(self._data_keys, obj=coil, grid=self._grid)
+
+        self._constants = {
+            "transforms": transforms,
+            "is_mixed_coils": is_mixed_coils,
+        }
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """Compute data of coil for given data key.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of the coil's degrees of freedom.
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self._constants.
+
+        Returns
+        -------
+        f : float or array of floats
+            Coil length.
+        """
+        if constants is None:
+            constants = self._constants
+
+        if constants["is_mixed_coils"]:
+            coils = self.things[0].coils
+            data = [
+                coils[i].compute(
```

Comment:
you can still use the coilset.compute method for mixed coilsets, you don't need to manually loop here

Reply:
I tried not to, but I manually looped because the compute function doesn't take a list of transforms as far as I know (and we want different transforms for each coil). Is there a way around that?

Reply:
The compute method for a mixed coilset should accept a list of transforms, or a single one which it uses for all of them

Reply:
Oh weird. I must've been doing something wrong before because it's working now. Thanks!